homeassistant/components/recorder/history/modern.py

```diff
@@ -254,6 +261,7 @@ def get_significant_states_with_session(
     start_time_ts = start_time.timestamp()
     end_time_ts = datetime_to_timestamp_or_none(end_time)
     single_metadata_id = metadata_ids[0] if len(metadata_ids) == 1 else None
+    lateral_join_for_start_time = instance.dialect_name == SupportedDialect.POSTGRESQL
```

Comment:
Can we somehow assert in tests that we set `lateral_join_for_start_time` as expected?

Reply:
Yeah... sadly I can't think of a good way other than patching....

Reply:
patching doesn't work
```
sqlalchemy.exc.InvalidRequestError: Can't invoke Python callable _significant_states_stmt() inside of lambda expression argument at <code object <lambda> at 0x1084c66a0, file "/Users/bdraco/home-assistant/homeassistant/components/recorder/history/modern.py", line 266>; lambda SQL constructs should not invoke functions from closure variables to produce literal values since the lambda SQL system normally extracts bound values without actually invoking the lambda or any functions within it.  Call the function outside of the lambda and assign to a local variable that is used in the lambda as a closure variable, or set track_bound_values=False if the return value of this function is used in some other way other than a SQL bound value.
```

Reply:
I ended up checking for LATERAL in the query logged by `sqlalchemy.engine`